astropy/modeling/core.py

```diff
@@ -3262,6 +3262,13 @@ def submodel_names(self):
                 newnames.append(item)
         return tuple(newnames)
 
+    @property
+    def submodel_classnames(self):
```

Comment:
Why the name and not the actual type?

Reply:
I happened to want the names, not opposed to returning types though. Perhaps I resurrect `submodel_types` to do what its name actually implies, and thus have properties for both?

Reply:
@Cadair , how would you use it if it returns the actual type?

Reply:
After thinking on this, it should return the actual types, not just the name/type-name. Then it can be used for `isinstance` checks and the like. If someone needs the name they can call it out of the type themselves.

Reply:
Hm, I expected `type(item)` here to return `<class 'astropy.modeling.functional_models.Linear1D'>` but instead it returns 
```
<class 'astropy.modeling.functional_models.Linear1D'>
Name: Linear1D
N_inputs: 1
N_outputs: 1
Fittable parameters: ('slope', 'intercept')
```
Was this a recent change in astropy? And are we fine returning this whole thing for each entry (looking more closely it's a `astropy.modeling.core._ModelMeta` object)?